prusti/src/callbacks.rs

```diff
@@ -91,6 +92,50 @@ impl prusti_rustc_interface::driver::Callbacks for PrustiCompilerCalls {
         });
     }
 
+    fn after_crate_root_parsing(&mut self, compiler: &Compiler, krate: &mut Crate) -> Compilation {
+        // `prusti-rustc` will pass a `--extern` option indicating where to find
+        // the `prusti-contracts` crate. This crate must be loaded in order to
+        // include the default extern specs (for e.g build-in `assert!`
+        // statements). However, if the user does not explicitly import the
+        // crate in the code or use an `extern crate` statement, then the
+        // compiler will not load the crate.
+        //
+        // To work around this, we add a dummy `extern crate` statement to the
+        // crate if one does not already exist.
+        //
+        // Ideally we could load the prusti-contracts crate directly in the
+        // TyCtxt rather than modifying the crate AST.  However, it does not
+        // seem possible to load the crate directly in the TyCtxt in the
+        // `after_expansion` or `after_analysis` callback as the `cstore` object
+        // appears to be frozen at that point.
+
+        if compiler.sess.opts.externs.get("prusti_contracts").is_some() {
+            let prusti_contracts_symbol = Symbol::intern("prusti_contracts");
+
+            if !krate.items.iter().any(|item| match item.kind {
+                ItemKind::ExternCrate(_, ident) => ident.name == prusti_contracts_symbol,
```

Comment:
[`ItemKind::ExternCrate`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_hir/hir/enum.ItemKind.html#variant.ExternCrate):

> An extern crate item, with optional original crate name if the crate was renamed.

So if one were to write `extern crate prusti_contracts as pc;`, we would miss this.

```suggestion
                ItemKind::ExternCrate(Some(original_ident), _) => original_ident == prusti_contracts_symbol,
                ItemKind::ExternCrate(None, ident) => ident.name == prusti_contracts_symbol,
```

Reply:
Nice catch, thanks!